### ISSUE TITLE
Hacktoberfest-2018 contributions and audits

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#662]
This issue is for tracking all `#hacktoberfest` contributions.
See https://hacktoberfest.digitalocean.com/eventkit.

## Step-by-step instruction:

1. Register at https://hacktoberfest.digitalocean.com/ by pressing "sign up" and using your github account. You should be able to see your progress at `https://hacktoberfest.digitalocean.com/stats/${username}` (e.g. https://hacktoberfest.digitalocean.com/stats/Quasilyte).

2. Choose Go project to contribute to. Pick any project you like and that seems like an active (last commit have been merged not too long ago). Let's say we've chosen `github.com/foo/bar` project (just an example).

3. Do a `go get -v github.com/foo/bar/...`. This downloads package under your `$GOPATH/src/github.com/foo/bar`.

4. Do a `cd $GOPATH/src/github.com/foo/bar` (for convenience).

4. Now run `gocritic` over the project. The suggested way is:

```bash
gocritic check-project -withExperimental -disable=unnamedResult,appendCombine,importPackageName,nestingReduce,emptyFmt,hugeParam,sqlRowsClose `pwd`
```
You may want to disable more checkers if they doesn't look useful to you.

5. Analyze the issues found on the (4) step. If there are some that may worth a PR, continue to the next step, otherwise start from the (2), now with a different project.

5.1 **(Optional)** If you've spotted an obvious false positive, please [fill an issue on gocritic tracker](https://github.com/go-critic/go-critic/issues/new). You can even try to fix that problem yourself, earning another contribution.

6. Fork a project on the github. Suppose your username is `gosu_pro`, then forked repo can have path like `github.com/gosu_pro/bar`.

7. From the `$GOPATH/src/github.com/foo/bar` directory do:

```bash
git remote add fork https://github.com/gosu_pro/bar.git
```

8. Now fix problems linter found. It's a good practice to concentrate on the most severe and single check first, packing it into a single commit, to make reviewers life easier. It's OK to send N PRs, every of which fixes only 1 kinds of a problem.

9. Create a branch with a fixes: `git checkout -b my_fix_1` (you can give a better branch name, for example, a checker name that found an issue).

10. Add and commit changes: `git add -u && git commit`. Write a good commit message. You may want to inspect previous repo commits, so your commit message is consistent with them. Or you can take some inspirations from the linked contributions inside this thread.

11. Push your changes to your fork: `git push fork my_fix_1`.

12. Now you can do a PR (Pull Request) on the github from your fork to the upstream (original) repo.
So, you need to merge `my_fix_1` branch into `master` (or `dev`, depends on the project, but maintainers can always redirect merge branch into something they find more appropriate).

13. **(Optional)** Write your contribution here! We will be glad to see how gocritic helps you to do open-source, etc.

====

**Registration**:

Please 

**Checking your progress**:

`https://hacktoberfest.digitalocean.com/stats/${username}`

For example,  `https://hacktoberfest.digitalocean.com/stats/Quasilyte`

**Comment by Quasilyte** [comment]
CC @cristaloleg @fexolm @ludweeg

**Comment by Quasilyte** [comment]
https://github.com/go-kit/kit/pull/767

**Comment by Quasilyte** [comment]
Looks like Microsoft did a good job encouraging even more contributors to participate.
Check out https://open.microsoft.com/2018/09/30/join-hacktoberfest-2018-celebration-microsoft/ and
https://twitter.com/BerndVerst/status/1046563359204622338
:tada:

**Comment by ludweeg** [comment]
| Status  |  Pull request  | Checker that found an issue  |
| :--------: | :------------: | :--------: |
| ? | grafeas/grafeas#226 | https://go-critic.github.io/overview#unslice-ref |

**Comment by cristaloleg** [comment]
go-gitea/gitea/pull/5115 `namedConst `
go-gitea/gitea/pull/5119 `defaultCaseOrder`
go-gitea/gitea/pull/5120 `sloppyLen `
go-gitea/gitea/pull/5121 `dupBranchBody `
go-gitea/gitea/pull/5122 `typeSwitchVar `

**Comment by cristaloleg** [comment]
See you on Hacktoberfest2019
